sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs

```diff
@@ -51,55 +51,45 @@ public sealed class StringLiteralShouldNotBeDuplicated : ParameterLoadingDiagnos
 
         protected override void Initialize(ParameterLoadingAnalysisContext context)
         {
-            context.RegisterCompilationStartAction(
-                csac =>
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
                 {
-                    if (csac.Compilation.IsTest())
-                    {
-                        return;
-                    }
+                    var stringLiterals = c.Node.DescendantNodes()
```

Comment:
Just a comment on readability: knowing the kind of the nodes the action is being registered against is important when reading and understanding the code. However, inlining the callback makes this hard to see; the reader has to just forward thirty or so lines to find the kind of nodes, then jump back again to continue reading.